packages/content-management/content_insights/content_insights_server/src/register.ts

```diff
@@ -0,0 +1,145 @@
+
+/**
+ * Configuration for the usage counter
+ */
+export interface ContentInsightsConfig {
+  /**
+   * e.g. 'dashboard'
+   * passed as a domainId to usage counter apis
+   */
+  domainId: string;
+
+  /**
+   * Can control created routes access via access tags
+   */
+  routeTags?: string[];
+
+  /**
+   * Retention period in days for usage counter data
+   */
+  retentionPeriodDays?: number;
+}
+
+export interface ContentInsightsDependencies {
+  usageCollection: UsageCollectionSetup;
+  http: CoreSetup['http'];
+  getStartServices: () => Promise<{
+    usageCollection: UsageCollectionStart;
+  }>;
+}
+
+export interface ContentInsightsStatsResponse {
+  result: ContentInsightsStats;
+}
+
+export interface ContentInsightsStats {
+  from: string;
+  daily: Array<{
+    date: string;
+    count: number;
+  }>;
+}
+
+/*
+ * Registers the content insights routes
+ */
+export const registerContentInsights = (
+  { usageCollection, http, getStartServices }: ContentInsightsDependencies,
+  config: ContentInsightsConfig
+) => {
+  const retentionPeriodDays = config.retentionPeriodDays ?? 90;
+  const counter = usageCollection.createUsageCounter(config.domainId, {
+    retentionPeriodDays,
+  });
+
+  const router = http.createRouter();
+  const validate = {
+    params: schema.object({
+      id: schema.string(),
+      eventType: schema.literal('viewed'),
+    }),
+  };
+  router.post(
+    {
+      path: `/internal/content_management/insights/${config.domainId}/{id}/{eventType}`,
+      validate,
+      options: {
+        tags: config.routeTags,
+      },
+    },
+    async (context, req, res) => {
+      const { id, eventType } = req.params;
+
+      counter.incrementCounter({
+        counterName: id,
+        counterType: eventType,
+        namespace: (await context.core).savedObjects.client.getCurrentNamespace(),
+      });
+      return res.ok();
+    }
+  );
+  router.get(
+    {
+      path: `/internal/content_management/insights/${config.domainId}/{id}/{eventType}/stats`,
+      validate,
+      options: {
+        tags: config.routeTags,
+      },
+    },
+    async (context, req, res) => {
+      const { id, eventType } = req.params;
+      const {
+        usageCollection: { search },
+      } = await getStartServices();
+
+      const subtractDaysFromToday = (date: Date, days: number) => {
+        const result = new Date(date);
+        result.setUTCDate(result.getUTCDate() - days);
+        return result;
+      };
+
+      const currentDate = new Date();
+      const startOfDay = new Date(
+        Date.UTC(currentDate.getUTCFullYear(), currentDate.getUTCMonth(), currentDate.getUTCDate())
+      );
+
+      const from = subtractDaysFromToday(startOfDay, retentionPeriodDays);
+
+      const result = await search({
+        filters: {
+          domainId: config.domainId,
+          counterName: id,
+          counterType: eventType,
+          namespace: (await context.core).savedObjects.client.getCurrentNamespace(),
```

Comment:
Not a big fan of forcing you to do this, I'll see if we can improve automatic namespace detection on our side easily.
Then we would simply have `useNamespace: true` instead.

This is by no means a blocker for this PR, let's use this strategy for now and try to improve on a follow-up.